testng-core/src/test/java/test/thread/MultiThreadedDependentTest.java

```diff
@@ -70,7 +71,8 @@ private void test(int threadCount) {
     tng.setThreadCount(threadCount);
     tng.setParallel(XmlSuite.ParallelMode.METHODS);
     Map<Long, Long> map = Helper.getMap(MultiThreadedDependentSampleTest.class.getName());
-    synchronized (map) {
+    KeyAwareAutoCloseableLock lock = new KeyAwareAutoCloseableLock();
+    try (KeyAwareAutoCloseableLock.AutoReleasable ignore = lock.lockForObject(map)) {
```

Comment:
This looks odd. If you allocate a fresh instance `new KeyAwareAutoCloseableLock`, then the locking is not very useful. Consider moving the instance to a field